sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/TransientErrorsRetryPolicy.scala

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.CosmosException
+import com.azure.cosmos.spark.diagnostics.BasicLoggingTrait
+
+import java.util.concurrent.atomic.AtomicLong
+import scala.util.Random
+import scala.util.control.Breaks
+
+private[spark] object TransientErrorsRetryPolicy extends BasicLoggingTrait {
+  private val rnd = Random
+
+  def executeWithRetry[T]
+  (
+    func: () => T,
+    initialMaxRetryIntervalInMs: Int = CosmosConstants.initialMaxRetryIntervalForTransientFailuresInMs,
+    maxRetryIntervalInMs: Int = CosmosConstants.maxRetryIntervalForTransientFailuresInMs,
+    maxRetryCount: Int = Int.MaxValue
```

Comment:
by default, we never gave up?

Reply:
Correct - why should we ever give up on transient errors? Same behavior as in Spark 2.4 - and same as 429 handling in Spark - we will retry forever.